src/app/lmem/matchingContext.ts

```diff
@@ -33,10 +36,30 @@ export const urlMatchesContext = (
   }
 };
 
-export const findMatchingOffersAccordingToPreferences = (
+export const filterContextsMatchingUrl = (
```

Comment:
@MaartenLMEM should we allow a specific notice to match only on a xpath with no domain/url specified ?  (Example : Any page containing «Maarten Samson» in a heading) ?

If yes, I suggest to change `urlMatchesContext` to :

```ts
export const urlMatchesContext = (
  url: string,
  { urlRegex, excludeUrlRegex }: MatchingContext
): boolean => {
  try {
    return (
      (!urlRegex || new RegExp(urlRegex, 'i').test(url)) && // <- here we allow bypassing `urlRegex`
      !(excludeUrlRegex && new RegExp(excludeUrlRegex, 'i').test(url))
    );
  } catch (err) {
    if (process.env.NODE_ENV !== 'test') {
      captureException(err);
    }
    return false;
  }
};
```

Warning : such notices may have an energy/performance impact thought, because xpath would be tested on every page seen …

Reply:
> @MaartenLMEM should we allow a specific notice to match only on a xpath with no domain/url specified ?  (Example : Any page containing «Maarten Samson» in a heading) ?
> 
> Warning : such notices may have an energy/performance impact thought, because xpath would be tested on every page seen …

Yes I think we should we allow a specific notice to match only on a xpath with no domain specified (or group of domains of course)
Because there are some cases we need to target websites that we don't know, for example fake sites that imitate a big brand site to get user Bank Card number et that uses some patterns that are recognizables in the page  content and impossible to predict url that will be used by this m... f....

Reply:
To be more precise, we’re talking about matching the page content (xpath) without limiting neither to a domain **nor an URL pattern**

Reply:
You are right @JalilArfaoui . I didn't mention _URL pattern_ in my last answer sentence, but yes it s the same conclusion : contributor can specify a page content part without domain nor an URL pattern.

Reply:
OK juste had a talk with @JalilArfaoui and I ^had to admit there is no current usecase with url matching (it s just maybe/probably in the future)
So @lutangar current state of code is ok

Reply:
Good to know :+1: 
I agree this feature could be useful, but again this might go too far. The URLs pattern thing